src/net/sourceforge/kolmafia/persistence/ModifierDatabase.java

```diff
@@ -0,0 +1,1524 @@
+      final ModifierType type, final String key, final String string) {
+    return parseModifiers(new Lookup(type, key), string);
+  }
+
+  public static final Modifiers parseModifiers(final Lookup lookup, final String string) {
+    Modifiers newMods = new Modifiers();
+
+    newMods.setLookup(lookup);
+
+    for (var mod : DoubleModifier.DOUBLE_MODIFIERS) {
+      Pattern pattern = mod.getTagPattern();
+      if (pattern == null) {
+        continue;
+      }
+
+      Matcher matcher = pattern.matcher(string);
+      if (!matcher.find()) {
+        continue;
+      }
+
+      if (matcher.group(1) != null) {
+        newMods.setDouble(mod, Double.parseDouble(matcher.group(1)));
+      } else {
+        newMods.addExpression(
+            new Indexed<>(mod, ModifierExpression.getInstance(matcher.group(2), lookup)));
+      }
+    }
+
+    for (var mod : BitmapModifier.BITMAP_MODIFIERS) {
+      Pattern pattern = mod.getTagPattern();
+      if (pattern == null) {
+        continue;
+      }
+
+      Matcher matcher = pattern.matcher(string);
+      if (!matcher.find()) {
+        continue;
+      }
+      int bitcount = 1;
+      if (matcher.groupCount() > 0) {
+        bitcount = StringUtilities.parseInt(matcher.group(1));
+      }
+      // bitmapMasks stores the next mask we're going to use for modifier mod
+      int mask = bitmapMasks.get(mod);
+      switch (bitcount) {
+        case 1 -> bitmapMasks.put(mod, mask << 1);
+        case 2 -> {
+          bitmapMasks.put(mod, mask << 2);
+          mask |= mask << 1;
+        }
+        default -> {
+          KoLmafia.updateDisplay(
+              "ERROR: invalid count for bitmap modifier in " + lookup.toString());
+          continue;
+        }
+      }
+      if (bitmapMasks.get(mod) == 0) {
+        KoLmafia.updateDisplay(
+            "ERROR: too many sources for bitmap modifier "
+                + mod.getName()
+                + ", consider using longs.");
+      }
+
+      newMods.addBitmap(mod, mask);
+    }
+
+    for (var mod : BooleanModifier.BOOLEAN_MODIFIERS) {
+      Pattern pattern = mod.getTagPattern();
+      if (pattern == null) {
+        continue;
+      }
+
+      Matcher matcher = pattern.matcher(string);
+      if (!matcher.find()) {
+        continue;
+      }
+
+      newMods.setBoolean(mod, true);
+    }
+
+    for (var mod : StringModifier.STRING_MODIFIERS) {
+      Pattern pattern = mod.getTagPattern();
+      if (pattern == null) {
+        continue;
+      }
+
+      Matcher matcher = pattern.matcher(string);
+      if (!matcher.find()) {
+        continue;
+      }
+
+      String value = matcher.group(1);
+
+      if (mod == StringModifier.CLASS) {
+        value = StringModifier.depluralizeClassName(value);
+      }
+
+      newMods.setString(mod, value);
+    }
+
+    newMods.setString(StringModifier.MODIFIERS, string);
+
+    return newMods;
+  }
+
+  // TODO: what's the difference between these are the above?
```

Comment:
I believe that these are not in the blue "enchantments" section of a desc file. But by all means, check it out. 🙂